src/lib.rs

```diff
@@ -209,45 +213,111 @@ pub fn read_versioned_msg<M: Message, R: Read>(
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 // Follow protocol definition: https://mavlink.io/en/guide/serialization.html#v1_packet_format
-pub struct MAVLinkV1MessageRaw {
-    pub start: u8, // = MAV_STX
-    pub payload_length: u8,
-    pub sequence: u8,
-    pub system_id: u8,
-    pub component_id: u8,
-    pub message_id: u8,
-    pub payload_buffer: [u8; 255],
-    pub checksum: u16,
+pub struct MAVLinkV1MessageRaw([u8; 1 + Self::HEADER_SIZE + 255 + 2]);
+
+impl Default for MAVLinkV1MessageRaw {
+    fn default() -> Self {
+        Self::new()
+    }
 }
 
 impl MAVLinkV1MessageRaw {
+    const HEADER_SIZE: usize = 5;
+
+    pub const fn new() -> Self {
```

Comment:
The `new` function is usually used when there are arguments to construct the object. The functionality should move to `default` and `new`could be removed.

Reply:
Ok!